main.nf

```
@@ -34,10 +34,18 @@ if (params.saige_base) {
   saige_base_ch = Channel
       .fromPath(params.saige_base, checkIfExists: true)
       .ifEmpty { exit 1, "SAIGE summary stats (base cohort) not found: ${params.saige}" }
-} else if (params.gwas_catalogue_base){
-  // gwas_catatologue_base_ch = Channel
-  //    .fromPath(params.gwas_catalogue_base, checkIfExists: true)
-  //    .ifEmpty { exit 1, "GWAS summary stats (base cohort) not found: ${params.gwas_catalogue_base}" }
+} else if (params.gwas_cat_study_id){
+  gwas_catalogue_ftp_ch = Channel
+      .fromPath(params.gwas_catalogue_ftp, checkIfExists: true)
+      .ifEmpty { exit 1, "GWAS catalogue ftp locations not found: ${params.gwas_catalogue_ftp}" }
+      .splitCsv(header: true)
+      .map { row -> tuple(row.study_accession, row.ftp_link_harmonised_summary_stats) }
+      .filter{ it[0] == params.gwas_cat_study_id}
+      .ifEmpty { exit 1, "The GWAS study accession number you provided does not come as a harmonized dataset that can be used as a base cohort "}
+      .flatten()
+      .last()
+} else {
```

This is equivalent to the more declarative:

```Suggestion
 if (!params.gwas_cat_study_id && !params.gwas_cat_study_id){
```

You can also reorder and check for this one 1st instead of 3rd in the if clauses, as conceptually it's we add fail checks first to fail early. See relevant reference from same Jenny Bryan talk [here](https://speakerdeck.com/jennybc/code-smells-and-feels?slide=49)

---

👍 Will be adding this change in next iterations (as I didn't click on commit suggestion before merging).